share/p2p/discovery/manager.go

```diff
@@ -0,0 +1,51 @@
+package discovery
+
+import (
+	"context"
+	"errors"
+)
+
+type Manager struct {
+	discs map[string]*Discovery
+
+	cancel context.CancelFunc
+}
+
+func NewManager(discs map[string]*Discovery) *Manager {
+	return &Manager{
+		discs: discs,
+	}
+}
+
+func (m *Manager) Start(ctx context.Context) error {
+	advertiseCtx, cancel := context.WithCancel(context.Background())
+	m.cancel = cancel
+
+	for _, d := range m.discs {
+		err := d.Start(ctx)
+		if err != nil {
+			log.Errorw("failed to start discovery", "err", err)
```

Comment:
Lets make an issue about stopping started discoveries if error happen

Reply:
We should make it for all services but yes.

Reply:
#3408

Reply:
Well, good to know that we don't stop node on fail. Problem with partially started discoveries will still persist, even if mentioned issue is implemented. So needs to land in mentioned/new issue or to be resolved with modified Stop here.